src/getQuerySchema.ts

```diff
@@ -0,0 +1,177 @@
+import { z } from "zod";
```

Comment:
This used to be `jsonQueryParser` and some manual validation. With zod schemas, we can be sure the query actually matches expectations. This may cause issues if people we're using features that we didn't explicitly support. If that does happen, we can consider adding to these schemas.